cmd/krel/cmd/ff_test.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"k8s.io/release/pkg/command"
+	"k8s.io/release/pkg/git"
+)
+
+const (
+	pseudoReleaseBranch = "release-2.0"
+	testCommitMessage   = "This is my test commit"
+)
+
+type sut struct {
+	repo        *git.Repo
+	baseDir     string
+	bareCopyDir string
+}
+
+func newSUT(t *testing.T) *sut {
+	// A local k/k repo will be our test base
+	tempDir := filepath.Join(os.TempDir(), "k8s-test-ff")
+	require.Nil(t, os.MkdirAll(tempDir, 0o755))
+
+	// The bare repo which is the pseudo remote base
+	bareDir := filepath.Join(tempDir, "bare")
+	const url = "https://github.com/kubernetes/kubernetes"
+	if _, err := os.Stat(bareDir); os.IsNotExist(err) {
+		require.Nil(t,
+			command.New("git", "clone", "--bare", url, bareDir).RunSuccess(),
+		)
+	}
+
+	now := time.Now().Unix()
+	bareCopyDir := filepath.Join(tempDir, fmt.Sprintf("bare-%d", now))
+	require.Nil(t, command.New("cp", "-r", bareDir, bareCopyDir).RunSuccess())
+
+	// The base repo where every test case is inherited
+	baseDir := filepath.Join(tempDir, fmt.Sprintf("base-%d", now))
+
+	// Clone the repo from the bare, which is safe to modify
+	require.Nil(t,
+		command.New("git", "clone", bareCopyDir, baseDir).RunSuccess(),
+	)
+
+	// Modify the bare repo with test content
+	require.Nil(t,
+		command.NewWithWorkDir(baseDir,
+			"git", "checkout", "-b", pseudoReleaseBranch,
+		).RunSuccess(),
+	)
+	require.Nil(t,
+		command.NewWithWorkDir(baseDir,
+			"git", "push", "-u", git.DefaultRemote, pseudoReleaseBranch,
+		).RunSuccess(),
+	)
+	require.Nil(t,
+		command.NewWithWorkDir(baseDir,
+			"git", "checkout", "master",
+		).RunSuccess(),
+	)
+	require.Nil(t,
```

Comment:
Looks like this requires configuring git user.

Reply:
Oops I see this is already being addressed in #1113 :)

Reply:
Yep, I think this should fix it.